msal/src/Microsoft.Identity.Client/Internal/Requests/AuthenticationRequestParameters.cs

```diff
@@ -84,22 +87,20 @@ internal class AuthenticationRequestParameters
                 }
                 else
                 {
-                    if (ClientCredential.Assertion == null || ClientCredential.ValidTo != 0)
+                    if (ClientCredential.Assertion == null || ClientCredential.ValidTo != 0 || ClientCredential.ContainsX5C != SendCertificate)
                     {
-                        bool assertionNearExpiry = (ClientCredential.ValidTo <=
-                                                    Jwt.JsonWebToken.ConvertToTimeT(DateTime.UtcNow +
-                                                                                    TimeSpan.FromMinutes(
-                                                                                    Constants 
-                                                                                    .ExpirationMarginInMinutes)));
```

Comment:
Seems from the code that we should also adhere to the ValidTo. This may be an external contract. The code should though be moved to the ClientCredentials class, to ensure separation of concern is maintained. 

You could add a bool to that class which is not serialized, but which returns whether the Assertion has expired. e.g. HasExpired

Reply:
#Resolved